helm/resource_release.go

```diff
@@ -414,14 +414,27 @@ func resourceReleaseCreate(d *schema.ResourceData, meta interface{}) error {
 
 	rel, err := client.Run(chart, values)
 
-	// Return error only if no release was created
-	// This will ensure we store even failed releases into the state
 	if err != nil && rel == nil {
 		return err
-	} else if err != nil && rel.Info.Status == release.StatusFailed {
+	}
+
+	if err != nil && rel != nil {
+		exists, existsErr := resourceReleaseExists(d, meta)
```

Comment:
There is a weird case you can run into where the above [Run](https://github.com/terraform-providers/terraform-provider-helm/pull/406/files#diff-70e42ed4d5009a76bddc6fb6c16b913bR415) function can return both an `error` and a `Release` object but the metadata for the release has not actually been persisted to a ConfigMap on the cluster. Here I am just explicitly checking that a release was indeed created.